gcloud-java-examples/src/main/java/com/google/gcloud/examples/BigQueryExample.java

```diff
@@ -0,0 +1,686 @@
+        while (iterator.hasNext()) {
+          System.out.println(iterator.next());
+        }
+      } else {
+        System.out.println("Query completed with errors. Errors:");
+        for (BigQueryError err : queryResponse.executionErrors()) {
+          System.out.println(err);
+        }
+      }
+    }
+
+    @Override
+    QueryRequest parse(String... args) throws Exception {
+      if (args.length == 1) {
+        return QueryRequest.of(args[0]);
+      }
+      throw new IllegalArgumentException();
+    }
+
+    @Override
+    protected String params() {
+      return "<query>";
+    }
+  }
+
+  static {
+    CREATE_ACTIONS.put("dataset", new CreateDatasetAction());
+    CREATE_ACTIONS.put("table", new CreateSimpleTableAction());
+    CREATE_ACTIONS.put("view", new CreateViewAction());
+    CREATE_ACTIONS.put("external-table", new CreateExternalTableAction());
+    INFO_ACTIONS.put("dataset", new DatasetInfoAction());
+    INFO_ACTIONS.put("table", new TableInfoAction());
+    INFO_ACTIONS.put("job", new JobInfoAction());
+    LIST_ACTIONS.put("datasets", new ListDatasetsAction());
+    LIST_ACTIONS.put("tables", new ListTablesAction());
+    LIST_ACTIONS.put("jobs", new ListJobsAction());
+    LIST_ACTIONS.put("data", new ListTableDataAction());
+    DELETE_ACTIONS.put("dataset", new DeleteDatasetAction());
+    DELETE_ACTIONS.put("table", new DeleteTableAction());
+    ACTIONS.put("create", new ParentAction(CREATE_ACTIONS));
+    ACTIONS.put("info", new ParentAction(INFO_ACTIONS));
+    ACTIONS.put("list", new ParentAction(LIST_ACTIONS));
+    ACTIONS.put("delete", new ParentAction(DELETE_ACTIONS));
+    ACTIONS.put("cancel", new CancelJobAction());
+    ACTIONS.put("load", new LoadAction());
+    ACTIONS.put("extract", new ExtractAction());
+    ACTIONS.put("copy", new CopyAction());
+    ACTIONS.put("query", new QueryAction());
+  }
+
+  private static void printUsage() {
+    StringBuilder actionAndParams = new StringBuilder();
+    for (Map.Entry<String, BigQueryAction> entry : ACTIONS.entrySet()) {
+      actionAndParams.append("\n\t").append(entry.getKey());
+
+      String param = entry.getValue().params();
+      if (param != null && !param.isEmpty()) {
+        actionAndParams.append(' ').append(param.replace("\n", "\n\t\t"));
+      }
+    }
+    System.out.printf("Usage: %s [<project_id>] operation [entity] <args>*%s%n",
+        BigQueryExample.class.getSimpleName(), actionAndParams);
+  }
+
+  @SuppressWarnings("unchecked")
+  public static void main(String... args) throws Exception {
+    BigQueryOptions.Builder optionsBuilder = BigQueryOptions.builder();
+    BigQueryAction action;
+    String actionName;
+    if (args.length >= 2 && !ACTIONS.containsKey(args[0])) {
+      actionName = args[1];
+      optionsBuilder.projectId(args[0]);
+      action = ACTIONS.get(args[1]);
+      args = Arrays.copyOfRange(args, 2, args.length);
+    } else {
+      actionName = args[0];
```

Comment:
If a user doesn't give any arguments, then this will cause an ArrayIndexOutOfBoundsException.  Instead of that, we should probably display the usage instructions.